react-hot-button-component/my-app/src/HotButton.js

```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from 'react';
+import './HotButton.css';
+
+export default function HotButton({ text, color }) {
+  // we are creating a state called clicks and initializing it to null with setClicks as the setter.
+  const [clicks, setClicks] = useState(null);
+  // passes an anynomous callback function where if clicks is truthy run the console.log
+  // this useEffect will only run when the state clicks is changed
+  useEffect(() => {
```

Comment:
Interesting. Did you learn this on your own? It's fun, but not really necessary since you could just log `clicks` right after line 6, without the `useEffect`. But this way, it only logs when the value changes.